src/getindex.jl

```diff
@@ -0,0 +1,113 @@
+const Indexer = AbstractVector{<:Integer}
+
+Base.IndexStyle(::LinearMap) = IndexCartesian()
+# required in Base.to_indices for [:]-indexing
+Base.eachindex(::IndexLinear, A::LinearMap) = Base.OneTo(length(A))
+Base.lastindex(A::LinearMap) = last(eachindex(IndexLinear(), A))
+Base.firstindex(A::LinearMap) = first(eachindex(IndexLinear(), A))
```

Comment:
I noticed that codecov complains about these two lines.  I wonder if they should be commented out because we are not supporting `A[1]` and `A[end]` at this point in time, right?